src/hu_stuff.c

```diff
@@ -1153,21 +1153,42 @@ static void HU_widget_build_sttime(void)
   int offset = 0;
   extern int time_scale;
 
-  if (time_scale != 100)
+  if (gameaction == ga_completed)
+    plr->btuse_tics = 0;
+
+  if (hud_level_time & HUD_WIDGET_HUD ||
+      (hud_level_time & HUD_WIDGET_AUTOMAP && automapactive))
   {
-    offset += sprintf(hud_timestr, "\x1b%c%d%% ",
-            '0'+CR_BLUE, time_scale);
+    if (time_scale != 100)
+    {
+      offset += sprintf(hud_timestr, "\x1b%c%d%% ",
+              '0'+CR_BLUE, time_scale);
+    }
+
+    if (totalleveltimes)
+    {
+      const int time = (totalleveltimes + leveltime) / TICRATE;
+
+      offset += sprintf(hud_timestr + offset, "\x1b%c%d:%02d ",
+              '0'+CR_GREEN, time/60, time%60);
+    }
+
+    offset += sprintf(hud_timestr + offset, "\x1b%c%d:%05.2f",
+                      '0'+CR_GRAY, leveltime/TICRATE/60, (float)(leveltime%(60*TICRATE))/TICRATE);
+
+    if (plr->btuse_tics)
+      offset += sprintf(hud_timestr + offset, " ");
   }
 
-  if (totalleveltimes)
+  if (plr->btuse_tics)
   {
-    const int time = (totalleveltimes + leveltime) / TICRATE;
-
-    offset += sprintf(hud_timestr + offset, "\x1b%c%d:%02d ",
-            '0'+CR_GREEN, time/60, time%60);
+    offset += sprintf(hud_timestr + offset, "\x1b%cU %02i:%05.02f",
+                      '0'+CR_GOLD,
```

Comment:
Or how about `CR_BLUE`? Just a matter of taste...

Reply:
I think the "time scale" part is blue already, and I personally prefer yellow because it stands out more. But, just like you said, it's a matter of taste.

Reply:
Right, we already have blue. And gold is a bit more eye-actching.